source/base/OwnedBuffer.ooc

```diff
@@ -0,0 +1,73 @@
+/*
+* Copyright (C) 2015 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use ooc-base
+import math
+
+OwnedBuffer: cover {
+	_pointer: UInt8*
+	_size: Int
+	_owner: Owner
+	pointer ::= this _pointer
+	size ::= this _size
+	init: func@ {
+		this _pointer = null
+		this _size = 0
+		this _owner = Owner Unknown
```

Comment:
Why initialize and then overwrite immediately? Covers are not zero the way objects are. But we should be calling another constructor.